keps/sig-cluster-lifecycle/kubeadm/2568-kubeadm-non-root-control-plane/README.md

```diff
@@ -0,0 +1,960 @@
+
+Then we update the kube-apiserver manifest as follows:-
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  annotations:
+    kubeadm.kubernetes.io/kube-apiserver.advertise-address.endpoint: 172.17.0.2:6443
+  creationTimestamp: null
+  labels:
+    component: kube-apiserver
+    tier: control-plane
+  name: kube-apiserver
+  namespace: kube-system
+spec:
+  securityContext:
+    supplementalGroups:
+    - 2100
+  containers:
+  - name: kube-apiserver
+    command:
+    - kube-apiserver
+    - --advertise-address=172.17.0.2
+    - --client-ca-file=/etc/kubernetes/pki/ca.crt
+    ... # omitted to save space 
+    securityContext:
+      runAsUser: 2000
+      runAsGroup: 2000
+      allowPrivilegeEscalation: false
+      seccompProfile:
+        type: runtime/default
+      capabilities:
+        drop:
+        - ALL
+    image: k8s.gcr.io/kube-apiserver:v1.21.0-beta.0.368_9850bf06b571d5-dirty
+    ... # omitted to save space
+    volumeMounts:
+    ... # omitted to save space
+    - mountPath: /etc/kubernetes/pki
+      name: k8s-certs
+      readOnly: true
+    ... # omitted to save space
+```
+
+Similarly kube-controller-manager's manifest would like:-
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  creationTimestamp: null
+  labels:
+    component: kube-controller-manager
+    tier: control-plane
+  name: kube-controller-manager
+  namespace: kube-system
+spec:
+  securityContext:
+    supplementalGroups:
+    - 2100
+  containers:
+  - name: kube-controller-manager
+    command:
+    - kube-controller-manager
+    - --cluster-signing-cert-file=/etc/kubernetes/pki/ca.crt
+    ...
+    # omitted to save space.
+    securityContext:
+      runAsUser: 2001
+      runAsGroup: 2001
+      allowPrivilegeEscalation: false
+      seccompProfile:
+        type: runtime/default
+      capabilities:
+        drop:
+        - ALL
+    image: k8s.gcr.io/kube-controller-manager:v1.21.0-beta.0.368_9850bf06b571d5-dirty
+    ... # omitted to save space.
+    volumeMounts:
+    - mountPath: /etc/kubernetes/pki
+      name: k8s-certs
+      readOnly: true
+    ... # omitted to save space. 
+```
+
+### Test Plan
+
+<!--
+**Note:** *Not required until targeted at a release.*
+
+Consider the following in developing a test plan for this enhancement:
+- Will there be e2e and integration tests, in addition to unit tests?
+- How will it be tested in isolation vs with other components?
+
+No need to outline all of the test cases, just the general strategy. Anything
+that would count as tricky in the implementation, and anything particularly
+challenging to test, should be called out.
+
+All code is expected to have adequate tests (eventually with coverage
+expectations). Please adhere to the [Kubernetes testing guidelines][testing-guidelines]
+when drafting this test plan.
+
+[testing-guidelines]: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
+-->
+
+### Graduation Criteria
```

Comment:
this section should not be empty.

BTW, @fabriziopandini and me discussed that we graduate this feature with a kubeadm FeatureGate.

Reply:
+1 on feature gate from my side, it'd allow us to opt in gradually and move over time the feature to be enabled by default once we're confident.